azure/services/natgateways/natgateways.go

```diff
@@ -18,156 +18,112 @@ package natgateways
-		if err != nil {
-			return errors.Wrapf(err, "failed to create nat gateway %s in resource group %s", natGatewaySpec.Name, s.Scope.ResourceGroup())
-		}
-		log.V(2).Info("successfully created nat gateway", "nat gateway", natGatewaySpec.Name)
-		natGateway := infrav1.NatGateway{
-			ID:   azure.NatGatewayID(s.Scope.SubscriptionID(), s.Scope.ResourceGroup(), natGatewaySpec.Name),
-			Name: natGatewaySpec.Name,
-			NatGatewayIP: infrav1.PublicIPSpec{
-				Name: natGatewaySpec.NatGatewayIP.Name,
-			},
+			// TODO: ideally we wouldn't need to set the subnet spec based on the result of the create operation
+			s.Scope.SetNatGatewayIDInSubnets(natGatewaySpec.ResourceName(), *natGateway.ID)
 		}
-		natGatewaySpec.Subnet.NatGateway = natGateway
-		s.Scope.SetSubnet(natGatewaySpec.Subnet)
-	}
-	return nil
-}
-
-func (s *Service) getExisting(ctx context.Context, spec azure.NatGatewaySpec) (*infrav1.NatGateway, error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "natgateways.Service.getExisting")
-	defer done()
-
-	existingNatGateway, err := s.Get(ctx, s.Scope.ResourceGroup(), spec.Name)
-	if err != nil {
-		return nil, err
-	}
-	// We must have a non-nil, non-"empty" PublicIPAddresses
-	if !(existingNatGateway.PublicIPAddresses != nil && len(*existingNatGateway.PublicIPAddresses) > 0) {
-		return nil, errors.Wrap(err, "failed to parse PublicIPAddresses")
-	}
-	// TODO do we want to eventually handle NatGateway resources w/ more than one public IP address?
-	// For now we assume the first one is the significant one
-	publicIPAddressID := to.String((*existingNatGateway.PublicIPAddresses)[0].ID)
-	resource, err := autorest.ParseResourceID(publicIPAddressID)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse Resource ID from PublicIPAddresses ID")
-	}
-	// We depend upon a non-empty ResourceName string
-	if resource.ResourceName == "" {
-		return nil, errors.Wrap(err, fmt.Sprintf("got unexpected ResourceName value from NatGateway PublicIpAddress, ResourceName=%s", resource.ResourceName))
 	}
 
-	return &infrav1.NatGateway{
-		ID:   to.String(existingNatGateway.ID),
-		Name: to.String(existingNatGateway.Name),
-		NatGatewayIP: infrav1.PublicIPSpec{
-			Name: resource.ResourceName,
-		},
-	}, nil
+	s.Scope.UpdatePutStatus(infrav1.NATGatewaysReadyCondition, serviceName, resultingErr)
+	return resultingErr
 }
 
-// Delete deletes the nat gateway with the provided name.
+// Delete deletes the NAT gateway with the provided name.
 func (s *Service) Delete(ctx context.Context) error {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "natgateways.Service.Delete")
 	defer done()
 
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
 	if !s.Scope.Vnet().IsManaged(s.Scope.ClusterName()) {
 		log.V(4).Info("Skipping nat gateway deletion in custom vnet mode")
+
+		s.Scope.UpdateDeleteStatus(infrav1.NATGatewaysReadyCondition, serviceName, nil)
```

Comment:
not sure we want to do this: if the NAT gateways aren't managed and we don't delete them, we shouldn't mark them as deleted in conditions

Reply:
Should we still set the put status in that case? And if users BYO VNet, should we care about updating the delete status to reflect if they delete the NAT gateway themselves?

Reply:
I'm leaning towards not showing any conditions at all for resources we don't manage and only focus the Status on resources we create/update/delete, but I think we need #1868 in order to do that cleanly